### PR TITLE
Allow selection of different units for temperature

### PR DESCRIPTION
**Pull request by phil-davis**
The API adds support for "unit" C, F or K in requests, and returns the
temperatures in those units Celcius, Farenheit, Kelvin respectively.
The Web App knows how to drive the enhanced API and has a new dropdown
control to select the units of measure.

I did this while @swalladge was doing other general stuff, so there are merge conflicts. I will look at those later. For now, putting the pull request here so the code can be reviewed.

### CODE REVIEW

File: database.py [hunk 1]
```diff
@@ -31,6 +31,24 @@ def dict(self):
                 'timestamp': self.timestamp
                 }
 
+class GeneralTemp(object):
```
Comment (swalladge) [inline database.py]: don't need to create a whole new object class for temperatures - we already have one!

**Comment by swalladge** [comment]
Going by the number of files with merge conflicts, it would probably be better if you could rebase your branch on the current master so we don't end up with a pile of messy merge commits.

File: database.py (hunk 1 above)
Comment (phil-davis) [inline database.py]: The problem I had was that the Temperature class was a database record thingy. When I was taking Temperature objects returned from database queries and modifying the temperature field to change from degrees C to F or K, I found that it seemed to be modifying the database itself. I end up with records that kept getting *1.8+32 and the number in the database got bigger and bigger every time I did an F or K unit query.
So I made a similar class that had ordinary fields in it. The code takes the data from the query results, puts it in this ordinary class and converts the temperatures.

**Comment by phil-davis** [comment]
Yes, once there is agreement about the way to go here, I will provide a rebased pull request.

File: database.py (hunk 1 above)
Comment (swalladge) [inline database.py]: What I was thinking is you could change the `dict` method signature to accept a
unit and perform the conversion there - then it's neat and out of the way (and
`dict` is called on every temperature object to format for sending as json).
Something like:

```python
def dict(self, units=None):
  if units=='F':
    temperature = self.temperature * <conversion factor>
  else:
    ...

  return { ...
```

Comment (swalladge) [inline database.py]: If you like I can implement it server-side and you can continue with the client implementation?

**Comment by phil-davis** [comment]
I edited the conflicts and got the diffs correct now, but it is still complaining. So I will make a whole new branch and put the total changes in it with a single commit. Then try out your idea of putting temperature conversion code into the dict() part - that seems a good idea and saves having to mess about with making an extra copy of the temperature records and then convert each of them. I will do that first thing in the morning.

**Comment by swalladge** [comment]
Ok, I'll close this pull request then.

**Pull request closed without merging**